### PR TITLE
[MODINVOICE-563] Encumbrance value incorrect after paying credit invoice

### PR DESCRIPTION
**Pull request by Saba-Zedginidze-EPAM**
## Purpose
[[MODINVOICE-563] Encumbrance value incorrect after paying credit invoice](https://folio-org.atlassian.net/browse/MODINVOICE-563)

## Approach
- Only update encumbrance amount if status is Unreleased

## Verification
image
image

### CODE REVIEW

**Review by azizbekxm — APPROVED** [review]

File: src/main/java/org/folio/service/transactions/batch/BatchPendingPaymentService.java [hunk 1]
```diff
@@ -89,9 +89,11 @@ private void updateEncumbranceToCancelTransaction(Transaction encumbrance, doubl
   }
 
   private void updateEncumbranceToApplyTransaction(Transaction encumbrance, double amount, CurrencyUnit currency) {
+    if (encumbrance.getEncumbrance().getStatus() == Encumbrance.Status.UNRELEASED) {
+      encumbrance.setAmount(subtractMoney(encumbrance.getAmount(), amount, currency));
+    }
     encumbrance.getEncumbrance().setAmountAwaitingPayment(sumMoney(
```
Comment (BKadirkhodjaev) [inline src/main/java/org/folio/service/transactions/batch/BatchPendingPaymentService.java:95]: Is it fine if we have these updated for Released encumbrance transactions?

Comment (BKadirkhodjaev) [reply src/main/java/org/folio/service/transactions/batch/BatchPendingPaymentService.java:95]: Something tells me retuning early on Released status may be applicable here, what do you think?

Comment (SerhiiNosko) [reply src/main/java/org/folio/service/transactions/batch/BatchPendingPaymentService.java:95]: I think it makes sense, at least we will keep the same pattern as for updateEncumbranceToCancelTransaction()
Probably this if was missed initially for this method.

**Review by SerhiiNosko — APPROVED** [review]

**Review by BKadirkhodjaev — APPROVED** [review]

File: src/main/java/org/folio/service/transactions/batch/BatchPendingPaymentService.java (hunk 1 above)
Comment (SerhiiNosko) [reply src/main/java/org/folio/service/transactions/batch/BatchPendingPaymentService.java:95]: Checked also our python script to fix finance issues.
We have command unrelease_open_orders_encumbrances_with_nonzero_amounts()
https://github.com/folio-org/mod-finance-storage/blob/master/scripts/fix_encumbrances/fix_encumbrances_quesnelia.py#L626

So it means that if encumbrance is released and has some non zero amount - it is not valid situation that is fixed by this script by unreleasing this encumbrance.

**Pull request merged**